modules/python/examples/ukf-nonlinear-complex-example.py

```diff
@@ -0,0 +1,536 @@
+  """
+  @brief Compute the commands realinp.sing a turn at constant linear velocity.
+
+  @param v Constant linear velocity.
+  @param angleStart Starting angle (in degrees).
+  @param angleStop Stop angle (in degrees).
+  @param nbSteps Number of steps to perform the turn.
+  @return List[ColVector] The corresponding list of commands.
+  """
+  cmds = []
+  dTheta = Math.rad(angleStop - angleStart) / float(nbSteps - 1);
+  for i in range(nbSteps):
+    theta = Math.rad(angleStart) + dTheta * float(i);
+    cmd = ColVector([v, theta])
+    cmds.append(cmd)
+  return cmds
+
+def generateCommands() -> List[ColVector]:
+  """
+  @brief Generate the list of commands for the simulation.
+
+  @return List[ColVector] The list of commands to use in the simulation
+  """
+  cmds = []
+  # Starting by an straight line acceleration
+  nbSteps = 30
+  dv = (1.1 - 0.001) / float(nbSteps - 1)
+  for i in range(nbSteps):
+    cmd = ColVector([0.001 + float(i) * dv, 0.])
+    cmds.append(cmd)
+
+  # Left turn
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, 0, 2, 15)
+  cmds.extend(leftTurnCmds)
+  for i in range(100):
+    cmds.append(cmds[len(cmds) -1])
+
+  # Right turn
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  rightTurnCmds = generateTurnCommands(lastLinearVelocity, 2, -2, 15);
+  cmds.extend(rightTurnCmds)
+  for i in range(200):
+    cmds.append(cmds[len(cmds) -1])
+
+  # Left  turn again
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, -2, 0, 15)
+  cmds.extend(leftTurnCmds)
+  for i in range(150):
+    cmds.append(cmds[len(cmds) -1])
+
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, 0, 1, 25)
+  cmds.extend(leftTurnCmds)
+  for i in range(150):
+    cmds.append(cmds[len(cmds) -1])
+
+  return cmds
+
+class vpBicycleModel:
+  """
+  @brief Class that approximates a 4-wheel robot unp.sing a bicycle model.
+  """
+  def __init__(self, w: float):
+    """
+    @brief Construct a new vpBicycleModel object.
+
+    @param w The length of the wheelbase.
+    """
+    self.m_w = w # The length of the wheelbase.
+
+  def computeMotion(self, u: ColVector, x: ColVector, dt: float) -> ColVector:
+    """
+    @brief Models the effect of the command on the state model.
+
+    @param u The commands. u[0] = velocity ; u[1] = steeringAngle .
+    @param x The state model. x[0] = x ; x[1] = y ; x[2] = heading
+    @param dt The period.
+    @return ColVector The state model after applying the command.
+    """
+    heading = x[2]
+    vel = u[0]
+    steeringAngle = u[1]
+    distance = vel * dt
+
+    if (abs(steeringAngle) > 0.001):
+      # The robot is turning
+      beta = (distance / self.m_w) * np.tan(steeringAngle)
+      radius = self.m_w / np.tan(steeringAngle);
+      np.sinh = np.sin(heading)
```

Comment:
np.sinh should be replaced with sinh = ...

Reply:
Done in e885c41ad